os_geometry/src/main.rs

```diff
@@ -0,0 +1,140 @@
+use std::f64::consts::PI;
+use std::fs::File;
+use std::io::{BufWriter, Cursor, Read, Write};
+
+use anyhow::Result;
+use geozero::mvt::{tile::Layer, Message};
+use indicatif::{ProgressBar, ProgressStyle};
+use mbtiles::Mbtiles;
+use pmtiles2::{util::tile_id, Compression, PMTiles, TileType};
+
+/// This script takes the `OSMasterMapTopography_gb_TopographicArea.mbtiles` file as input, filters for features with the "Roads Tracks And Paths" theme, and re-encodes as pmtiles.
+#[tokio::main]
+async fn main() -> Result<()> {
+    let args: Vec<_> = std::env::args().collect();
+    let path = &args[1];
+    println!("Opening {path}");
+
+    let mbtiles = Mbtiles::new(path)?;
+    let mut conn = mbtiles.open_readonly().await?;
+
+    let metadata = mbtiles.get_metadata(&mut conn).await?;
+    let bounds = metadata.tilejson.bounds.unwrap();
+    // TODO Could read as the minzoom/maxzoom, which should be the same
+    let zoom = 16_u8;
+    let (x1, y1) = lon_lat_to_tile(bounds.left, bounds.top, zoom.into());
+    let (x2, y2) = lon_lat_to_tile(bounds.right, bounds.bottom, zoom.into());
+    println!("Reading x = {x1} to {x2}, y = {y1} to {y2}");
+
+    let mut pmtiles = PMTiles::new(TileType::Mvt, Compression::GZip);
+    pmtiles.min_longitude = bounds.left;
+    pmtiles.min_latitude = bounds.bottom;
+    pmtiles.max_longitude = bounds.right;
+    pmtiles.max_latitude = bounds.top;
+    pmtiles.min_zoom = zoom;
+    pmtiles.max_zoom = zoom;
+    pmtiles.meta_data.insert(
+        "vector_layers".to_string(),
+        serde_json::json!([
+            {
+                "id": "topographic_area",
+                "minzoom": zoom,
+                "maxzoom": zoom,
+                "fields": {
+                    "toid": "String",
```

Comment:
There are more fields here, but actually it might not be required to set this metadata for use in maplibre...